### PR TITLE
[FIX] l10n_in: consider discounted price for HSN summary

### PR DESCRIPTION
**Pull request by niai-odoo**
*Before this commit:*
- The HSN summary was calculated based on the unit price of the line items, even if a discount was applied.

*After this commit:*
- The HSN summary is now calculated based on the discounted price of the line items.

opw: 4040478

### CODE REVIEW

**Comment by h4818** [comment]
Not sure about the rounding, but everything else LGTM
@robodoo r+

**Comment by h4818** [comment]
@smetl should we include `discount` in the base lines dict and change 
`_l10n_in_get_hsn_summary_table` in both python and js?

**Comment by h4818** [comment]
@robodoo r-
shouldn't this change too? https://github.com/odoo/odoo/blob/saas-17.2/addons/l10n_in_pos/static/src/overrides/models/order.js#L40

**Comment by niai-odoo** [comment]
@h4818 

Thank you for pointing out the issue with the `l10n_in_pos` module. I have fixed it in the POS as well.

File: addons/l10n_in/models/account_invoice.py [hunk 1]
```diff
@@ -188,7 +188,7 @@ def _l10n_in_get_hsn_summary_table(self):
             base_lines.append({
                 'l10n_in_hsn_code': line.l10n_in_hsn_code,
                 'quantity': line.quantity,
-                'price_unit': line.price_unit,
+                'price_unit': line.currency_id.round(line.price_unit * (1 - (line.discount / 100.0))),
```
Comment (smetl) [inline addons/l10n_in/models/account_invoice.py:191]: You must not do something on one language that is not in the other -_- if you round python-side, you need to round js-side too. I don't think you want to loose some extra precision here.

File: addons/l10n_in/tests/test_hsn_summary.py [hunk 2]
```diff
@@ -578,3 +579,68 @@ def test_l10n_in_hsn_summary_4(self):
             },
         ))
         self._assert_tests(tests)
+
+    def test_l10n_in_hsn_summary_5(self):
+        """ Test with discount on lines. """
+        move = self.env['account.move'].create({
```
Comment (smetl) [inline addons/l10n_in/tests/test_hsn_summary.py:585]: The whole test is designed to test both python and javascript side so you must absolutely not create an invoice like this.

Comment (niai-odoo) [reply addons/l10n_in/tests/test_hsn_summary.py:585]: I agree that this test is designed to test both the Python and JavaScript sides. However, the problem is that the value going into the method `_l10n_in_get_hsn_summary_table` is wrong, which is why I created the invoice, so the method `_l10n_in_get_hsn_summary_table` gets called.

Please let me know if there is another way I can check without creating an invoice.

File: addons/l10n_in/models/account_invoice.py (hunk 1 above)
Comment (niai-odoo) [reply addons/l10n_in/models/account_invoice.py:191]: I have added rounding in js-side too

**Comment by jva-odoo** [comment]
Hello @smetl 
The customer is asking for an update so plz process this from your end

File: addons/l10n_in/tests/test_hsn_summary.py (hunk 2 above)
Comment (smetl) [reply addons/l10n_in/tests/test_hsn_summary.py:585]: @niai-odoo Add the discount to the lines provided to the method then :)

File: addons/l10n_in/models/account_invoice.py (hunk 1 above)
Comment (smetl) [reply addons/l10n_in/models/account_invoice.py:191]: @niai-odoo But are you sure about your rounding here? If the user decided to express the price unit using a decimal precision of 5 digits, I think you have to use 5 digits here, not 2

**Comment by niai-odoo** [comment]
Hello @smetl,

I recently saw the PR addressing this same issue. Which PR would you prefer to continue working on?
https://github.com/odoo/odoo/pull/173494

**Comment by smetl** [comment]
@niai-odoo https://github.com/odoo/odoo/pull/173494 seems closer to the merge.

**Pull request closed without merging**